client/my-sites/stats/features/modules/stats-utm/stats-module-utm.jsx

```diff
@@ -88,17 +115,35 @@ const StatsModuleUTM = ( {
 	const displaySummaryLink = data && ! hideSummaryLink;
 	const showLoader = isLoading || isFetchingUTM;
 
-	const getHref = () => {
-		if ( ! hideSummaryLink && summaryUrl ) {
-			return summaryUrl;
-		}
-		// Some modules do not have view all abilities
-		if ( ! summary && period && path && siteSlug ) {
-			return `/stats/${ period.period }/${ path }/${ siteSlug }?startDate=${ period.startOf.format(
-				'YYYY-MM-DD'
-			) }`;
-		}
-	};
+	const getHref = useMemo( () => {
+		return () => {
+			const params = new URLSearchParams( url.params.toString() );
```

Comment:
A unified way to deal with Odyssey Stats parameters is to read params from context.